src/test_lib_json/main.cpp

```diff
@@ -307,7 +307,21 @@ JSONTEST_FIXTURE(ValueTest, arrayIssue252) {
   }
   // JSONTEST_ASSERT_EQUAL(5, root["array"].size());
 }
+JSONTEST_FIXTURE(ValueTest, arrayIssue691) {
+  Json::Value array2;
+  array2.append(10);  // index 0
+  array2.append(20);  // index 1
+  array2.append(30);  // index 2
+  array2.append(50);  // index 3
 
+  JSONTEST_ASSERT_EQUAL(Json::Value(50),array2[3]);
+
+  array2.insert(3,40); // index 3
```

Comment:
Need to test the insert of rvalues and lvalues to make sure both overloads are implemented correctly.